examples/qml_minimal/CMakeLists.txt

```diff
@@ -78,20 +82,37 @@ add_library(${APP_NAME}_lib INTERFACE)
 # overwriting another's files.
 target_include_directories(${APP_NAME}_lib INTERFACE "${CXXQT_EXPORT_DIR}/${CRATE}")
 
-target_link_libraries(${APP_NAME}_lib INTERFACE
-    # WHOLE_ARCHIVE is needed for the generated QML plugin to register on startup,
-    # otherwise the linker will discard the static variables that initialize it.
-    "$<LINK_LIBRARY:WHOLE_ARCHIVE,${CRATE}-static>"
-    Qt::Core
-    Qt::Gui
-    Qt::Qml
-    Qt::QuickControls2
-)
+if(BUILD_WASM)
+    target_link_libraries(${APP_NAME}_lib INTERFACE
+        Qt::Core
+        Qt::Gui
+        Qt::Qml
+        Qt::QuickControls2
+    )
+    
+    target_link_options(${APP_NAME}_lib INTERFACE 
+        "SHELL:-Wl,--whole-archive $<TARGET_FILE:${CRATE}-static> -Wl,--no-whole-archive"
+    )
```

Comment:
So because the library here is not in the `target_link_libraries` it isn't marked as a dependency of the target, so it isn't always built before the target reaches linking. There are a few options here

# Add dependency

Simplest is add a dependency until we figure out how to make `LINK_LIBRARY:WHOLE_ARCHIVE` work

```cmake
    # Ensure that the crate has been built to link against
    add_dependencies(${APP_NAME}_lib ${CRATE}-static)

    # FIXME: "$<LINK_LIBRARY:WHOLE_ARCHIVE,${CRATE}-static>" fails with the following
    #
    # Feature 'WHOLE_ARCHIVE', specified through generator-expression
    # '$<LINK_LIBRARY>' to link target 'example_qml_minimal', is not supported
    # for the 'CXX' link language.
    #
    # Does a newer version of Qt/emscripten solve this?
    target_link_options(${APP_NAME}_lib INTERFACE
        "SHELL:-Wl,--whole-archive $<TARGET_FILE:${CRATE}-static> -Wl,--no-whole-archive"
    )
```

# Add to target_link_libraries

We could potentially have the whole-archive args in the target_link_libraries.

```cmake
    target_link_libraries(${APP_NAME}_lib INTERFACE
        Qt::Core
        Qt::Gui
        Qt::Qml
        Qt::QuickControls2
        # WHOLE_ARCHIVE is needed for the generated QML plugin to register on startup,
        # otherwise the linker will discard the static variables that initialize it.
        -Wl,--whole-archive
        $<TARGET_PROPERTY:${CRATE}-static,INTERFACE_LINK_LIBRARIES>
        -Wl,--no-whole-archive
    )
```

Note that this would likely also work for non-WASM builds but is not preferred and `"$<LINK_LIBRARY:WHOLE_ARCHIVE,${CRATE}-static>"` should be used as that uses generators etc, we should try to figure out why this doesn't work with wasm first :thinking: 

# Mixture of link library and options

It seems possible also to specify the library and a link library then override the link options after.

Again seems better to figure out why the cmake generator route doesn't work.

```cmake
    target_link_libraries(${APP_NAME}_lib INTERFACE
        Qt::Core
        Qt::Gui
        Qt::Qml
        Qt::QuickControls2
        ${CRATE}-static
    )

    # WHOLE_ARCHIVE is needed for the generated QML plugin to register on startup,
    # otherwise the linker will discard the static variables that initialize it.
    target_link_options(${APP_NAME}_lib INTERFACE
        "SHELL:-Wl,--whole-archive $<TARGET_FILE:${CRATE}-static> -Wl,--no-whole-archive"
    )
```

Reply:
See other comment below about using `CMAKE_LINK_LIBRARY_USING_WHOLE_ARCHIVE` instead, this is now not relevant.